### PR TITLE
Ability to add operator keys using Terraform

### PR DESCRIPTION
**Pull request by dharsanb**
Resource to manage [Keys](https://api-docs.humanitec.com/#tag/PublicKeys).

### CODE REVIEW

File: internal/provider/resource_keys_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	key := getPublicKey(t)
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read testing
+			{
+				Config: testAccResourceKey(key),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("humanitec_key.key_test", "key", key),
```
Comment (mateuszjenek) [inline internal/provider/resource_keys_test.go:21]: I would add more checks here e.g. if the id attribute is set

File: internal/provider/provider.go [hunk 2]
```diff
@@ -161,6 +161,7 @@ func (p *HumanitecProvider) Resources(ctx context.Context) []func() resource.Res
 		NewResourceDefinitionResource,
```
Comment (johanneswuerbach) [inline internal/provider/provider.go:161]: Unrelated to this line, could you add an example similar to https://github.com/humanitec/terraform-provider-humanitec/tree/main/examples/resources/humanitec_agent and run `go generate ./...` to update the docs?

**Review by mateuszjenek — APPROVED** [review]
One minor thing. The implementation looks good 🚀

File: internal/provider/resource_keys.go [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+	orgId  string
+}
+
+// OperatorKeyModel describes the key data model.
+type OperatorKeyModel struct {
+	ID          types.String   `tfsdk:"id"`
+	Key         types.String   `tfsdk:"key"`
+	CreatedAt   types.String   `tfsdk:"created_at"`
+	CreatedBy   types.String   `tfsdk:"created_by"`
+	ExpiredAt   types.String   `tfsdk:"expired_at"`
```
Comment (johanneswuerbach) [inline internal/provider/resource_keys.go:42]: For other resources we don't expose those status flags yet, could we remove them until somebody asks for them?

File: internal/provider/resource_keys.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+		MarkdownDescription: "A key is used by Humanitec to ensure ensure access to Humanitec hosted drivers. The key helps Humanitec operator to establish identity against the Humanitec Driver API",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The ID which refers to a specific key.",
+				Computed:            true,
+			},
+			"key": schema.StringAttribute{
+				MarkdownDescription: "The public key that is used for authentication.",
+				Required:            true,
```
Comment (johanneswuerbach) [inline internal/provider/resource_keys.go:62]: Could you add:
```
PlanModifiers: []planmodifier.String{
  stringplanmodifier.RequiresReplace(),
},
```

here, so Terraform knows that keys can't be updated?

**Review by johanneswuerbach — COMMENTED** [review]
Looks good already, a few minor comments.

**Review by johanneswuerbach — APPROVED** [review]

**Pull request merged**